src/NHibernate/Linq/Visitors/HqlGeneratorExpressionVisitor.cs

```diff
@@ -206,6 +222,18 @@ private HqlTreeNode VisitNhNominated(NhNominatedExpression nhNominatedExpression
 
 		private HqlTreeNode VisitInvocationExpression(InvocationExpression expression)
 		{
+			//This is an ugly workaround for dynamic expressions.
```

Comment:
In .NET Core/Mono the `DynamicExpression` is "reduced" to a CallSite delegate call, but on full .NET it does not happen.